src/wutils_db.erl

```diff
@@ -90,8 +95,10 @@ query(Query, PG) ->
             V
     catch
         exit:connection_timeout ->
+            close(PG),
```

Comment:
This seems reasonable but beware that this will only work if you have one pool (`primary`). I never found it important to have multiple pools, but I think it's worth noting.

To make it work with multiple pools, I think I'd make `open/0,1` return a record holding a pool and a connection for `close/1,2` to work with. The `run` and `query` functions would also need to be modified to unwrap that record and pass the bare connection to `pgsql` but then they'd know what pool to talk to when they were trying to close a connection.

Reply:
👍